ui/core/components/sim_header.ts

```diff
@@ -0,0 +1,296 @@
+				resp.json()
+					.then((versionInfo) => {
+						if (versionInfo.outdated == 2) {
+              downloadBinary.setAttribute('data-bs-title', 'Newer version of simulator available for download')
+							downloadBinary.classList.add('link-danger')
+              new Tooltip(downloadBinary);
+              this.addToolbarItem(downloadBinary);
+						}
+					})
+					.catch(error => {
+						console.warn('No version info found!');
+					});
+			});
+		} else {
+      downloadBinary.setAttribute('data-bs-title', 'Download simulator for faster simulating')
+      new Tooltip(downloadBinary);
+      this.addToolbarItem(downloadBinary);
+		}
+  }
+
+  private addSimOptionsLink() {
+		let optionsFragment = document.createElement('fragment');
+		optionsFragment.innerHTML = `
+			<a
+				href="javascript:void(0)"
+				class="sim-options"
+				role="button"
+				onmousedown="${onclick}"
+				data-bs-toggle="tooltip"
+				data-bs-placement="bottom"
+				data-bs-title="Show Sim Options"
+			>
+				<i class="fas fa-cog fa-lg"></i>
+			</a>
+		`;
+		let optionsLink = optionsFragment.children[0] as HTMLElement;
+		optionsLink.addEventListener('click', () => new SettingsMenu(this.rootElem, this.simUI));
+		new Tooltip(optionsLink);
+		this.addToolbarItem(optionsLink);
+	}
+
+	private addDiscordLink() {
+		let discordFragment = document.createElement('fragment');
+		discordFragment.innerHTML = `
+			<a
+				href="https://discord.gg/p3DgvmnDCS"
+				target="_blank"
+				class="link-alt discord-link"
+				data-bs-toggle="tooltip"
+				data-bs-placement="bottom"
+				data-bs-title="Join us on Discord"
+			>
+				<i class="fab fa-discord fa-lg"></i>
+			</a>
+		`;
+
+		let discordLink = discordFragment.children[0] as HTMLElement;
+		new Tooltip(discordLink);
+		this.addToolbarSocial(discordLink);
+	}
+
+	private addGitHubLink() {
+		let githubFragment = document.createElement('fragment');
+		githubFragment.innerHTML = `
+			<a
+				href="https://github.com/wowsims/wotlk"
+				target="_blank"
+				class="link-alt github-link"
+				data-bs-toggle="tooltip"
+				data-bs-placement="bottom"
+				data-bs-title="Contribute on GitHub"
+			>
+				<i class="fab fa-github fa-lg"></i>
+			</a>
+		`;
+
+		let githubLink = githubFragment.children[0] as HTMLElement;
+		new Tooltip(githubLink);
+		this.addToolbarSocial(githubFragment.children[0] as HTMLElement);
+	}
+
+	private addPatreonlink() {
+		let patreonFragment = document.createElement('fragment');
+		patreonFragment.innerHTML = `
+			<a href="https://patreon.com/wowsims" target="_blank" class="link-alt patreon-link">
+				<i class="fab fa-patreon fa-lg"></i>
+				<span>Support our devs</span>
+			</a>
+		`;
+
+		let patreonLink = patreonFragment.children[0] as HTMLElement;
+		new Tooltip(patreonLink);
+		this.addToolbarSocial(patreonFragment.children[0] as HTMLElement);
+	}
+}
+
+let headerFragment = document.createElement('fragment');
```

Comment:
I can't seem to find a way to move this into the class. The way the components are set up isn't really conducive to it. You can't initialize it as a property outside the constructor, you can't make it a method because you can't use `this` before calling super (and you have to pass the element to super to use the root element)